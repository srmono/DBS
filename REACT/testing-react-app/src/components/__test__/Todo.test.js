import {render, screen, cleanup} from '@testing-library/react';
import Todo from '../Todo';

afterEach( () => {
    cleanup()
})

test("Should render non-completed Todo", () => {
    const todo = { id: 1, title: "Git practice", completed: false}
    render(<Todo todo={todo} />); // Render Component
    const todoElement = screen.getByTestId('todo-1'); // select element
     // Assertions
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("Git practice");
    expect(todoElement).not.toContainHTML('<strike>');
})

test("Should render completed Todo", () => {
    const todo = { id: 2, title: "React Practice", completed: true}
    render(<Todo todo={todo} />); // Render Component
    const todoElement = screen.getByTestId('todo-2'); // select element
     // Assertions
    expect(todoElement).toBeInTheDocument();
    expect(todoElement).toHaveTextContent("React Practice");
    expect(todoElement).toMatchSnapshot('<strike>');
})