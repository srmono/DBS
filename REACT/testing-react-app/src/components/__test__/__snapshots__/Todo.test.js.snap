// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render completed Todo: <strike> 1`] = `
<div
  data-testid="todo-2"
>
  <strike>
     
    <h1>
      React Practice
    </h1>
     
  </strike>
</div>
`;
